.github/workflows/CI.yml

```diff
@@ -170,11 +171,18 @@ jobs:
                ${{ matrix.extra }}
                if [[ -n ${NO_COVERAGE} ]] ; then
                   echo "No coverage reporting"
-               else
-                  echo "Coverage reporting, updating compiler and linker flags"
+               elif [[ -n ${SLOW_COVERAGE_FLAGS} ]] ; then
```

Comment:
The addition of `SLOW_COVERAGE_FLAGS` to `extra` in two test cases means that those CI tests change their names, and the branch protection rules need to be adjusted to those names. I can do that, but I wonder: those flags are meant to be used on macOS or Windows, right? Wouldn't it then perhaps make more sense to drop `SLOW_COVERAGE_FLAGS` and just check for macOS or Windows (or even "not Linux") here. It may be less flexible (but we don't know that we'll ever need flexibility here), but on the upside we can't forget to use it if we add more jobs... Thoughts?

Reply:
That would be fine. My initial process here was just to see where I needed the SLOW_COVERAGE_FLAGS options (I didn't know is some linux builds might need it for some weird reason), but indeed it's ended up just being windows and mac.

Reply:
I'll push this change and lets the tests cycle (with the pthread commit), then I'll remove the pthread commit for merging (as I see it had some changes made to it)

Reply:
Didn't notice you made the changes here, I've now added them back and pushed